.github/workflows/pr.yaml

```diff
@@ -105,50 +72,43 @@ jobs:
   test:
     name: Test with Node.js ${{ matrix.node-version }}
     runs-on: ubuntu-latest
-    needs: check-image
-    if: always()  # Sempre executa o job
     strategy:
       matrix:
         node-version: [22]
     steps:
       - name: Checkout
-        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+        uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
+      - name: Check Image and Relevant Changes
+        id: check-image
+        uses: ./.github/actions/check-image-and-changes
+
       - name: Check if tests should be skipped
         run: |
-          if [[ "${{ needs.check-image.outputs.relevant_changes }}" == "false" && "${{ needs.check-image.outputs.image_exists }}" == "true" ]]; then
+          if [[ "${{ steps.check-image.outputs.relevant_changes }}" == "false" && "${{ steps.check-image.outputs.image_exists }}" == "true" ]]; then
             echo "Skipping tests: No relevant changes detected."
             exit 0
           fi
 
       - name: Setup Node.js
-        uses: actions/setup-node@cdca7365b2dadb8aad0a33bc7601856ffabcc48e # v4
+        uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.node-version }}
           registry-url: 'https://registry.npmjs.org'
 
       - name: Setup local Turbo cache
-        uses: dtinth/setup-github-actions-caching-for-turbo@cc723b4600e40a6b8815b65701d8614b91e2669e # v1
+        uses: dtinth/setup-github-actions-caching-for-turbo@v1
 
       - name: Use app-config.example.yaml
         run: rm app-config.yaml && mv app-config.example.yaml app-config.yaml
 
       - name: Install dependencies
-        uses: backstage/actions/yarn-install@b3c1841fd69e1658ac631afafd0fb140a2309024 # v0.6.17
+        uses: backstage/actions/yarn-install@v0.6.17
         with:
           cache-prefix: ${{ runner.os }}-v${{ matrix.node-version }}
 
-      - name: Run monorepo tools
```

Comment:
why did you remove this and the dockerfile regen check? please explain in the issue description -- this is a new change today that was not in the original PR, where you only removed the prettier and linter checks. 

https://github.com/redhat-developer/rhdh/pull/2575/commits/373dac41c690909770704703f9ebfc1b09a30a52#diff-1eb4e5fd5611777d4e597ef299a1cb5ba8050c28a2dabbd4fbc56205d69e5ddaL90

Reply:
Apologies for the removal, it was unintentional. I got a bit lost during the refactoring while juggling the test day and updating the PR at the same time. In the process, I mistakenly removed the Dockerfile regeneration check. I've now updated the PR and restored everything that was removed. Thanks for pointing it out!